src/sentry/migrations/0309_fix_many_to_many_field.py

```diff
@@ -0,0 +1,117 @@
+                ),
+                migrations.CreateModel(
+                    name="DashboardProject",
+                    fields=[
+                        (
+                            "id",
+                            sentry.db.models.fields.bounded.BoundedBigAutoField(
+                                primary_key=True, serialize=False
+                            ),
+                        ),
+                    ],
+                    options={
+                        "db_table": "sentry_dashboardproject",
+                    },
+                ),
+                migrations.AlterField(
+                    model_name="authprovider",
+                    name="default_teams",
+                    field=models.ManyToManyField(
+                        blank=True, through="sentry.AuthProviderDefaultTeams", to="sentry.Team"
+                    ),
+                ),
+                migrations.AlterField(
+                    model_name="dashboard",
+                    name="projects",
+                    field=models.ManyToManyField(
+                        through="sentry.DashboardProject", to="sentry.Project"
+                    ),
+                ),
+                migrations.AddField(
+                    model_name="dashboardproject",
+                    name="dashboard",
+                    field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="sentry.Dashboard"
+                    ),
+                ),
+                migrations.AddField(
+                    model_name="dashboardproject",
+                    name="project",
+                    field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="sentry.Project"
+                    ),
+                ),
+                migrations.AddField(
+                    model_name="authproviderdefaultteams",
+                    name="authprovider",
+                    field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="sentry.AuthProvider"
+                    ),
+                ),
+                migrations.AddField(
+                    model_name="authproviderdefaultteams",
+                    name="team",
+                    field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="sentry.Team"
+                    ),
+                ),
+                migrations.AlterUniqueTogether(
+                    name="dashboardproject",
+                    unique_together={("project", "dashboard")},
+                ),
+                migrations.AlterUniqueTogether(
+                    name="authproviderdefaultteams",
+                    unique_together={("authprovider", "team")},
+                ),
+            ],
+        )
+    ]
```

Comment:
I just realised - we should have the ALTER statements here to convert self-hosted and ST customer to have bigint columns here on these tables as well. Otherwise they will remain as `int` 

This also means we should mark this as dangerous, so that it doesn't run in prod (since I believe we've already fixed these columns there)